src/appwindow.cpp

```diff
@@ -58,6 +58,24 @@ AppWindow::AppWindow(bool noHotExit, QWidget *parent) : QMainWindow(parent), ui(
     allocate();
     setConnections();
 
+    auto separator = ui->menuFile->insertSeparator(ui->actionSave); // used to insert openRecentFilesMenu
+    auto openRecentFilesMenu = new QMenu(tr("Open Recent Files"), ui->menuFile);
+    ui->menuFile->insertMenu(separator, openRecentFilesMenu);
+    connect(openRecentFilesMenu, &QMenu::aboutToShow, [this, openRecentFilesMenu] {
+        openRecentFilesMenu->clear();
+        if (SettingsHelper::getRecentFiles().isEmpty())
+        {
+            openRecentFilesMenu->addAction(tr("No file is opened recently"))->setDisabled(true);
```

Comment:
```suggestion
            openRecentFilesMenu->addAction(tr("No Recent Files"))->setDisabled(true);
```

It seems more short and straightforward.

Reply:
I tried to explain what "recent file" is.

Reply:
Isn't word **recent files** self explanatory?

Reply:
It's not necessary to explain "recent files", but here, I think we can explain it via a warning. Though it's not necessary, it's not completely meaningless. "Recent files" doesn't point out what is "recent". Recently edited? Recently opened? Recently used elsewhere instead of in CP Editor?

Reply:
What's more, "No recent files" is a noun phrase instead of a sentence, which sounds weird to me. I think "There is no recent file" is much better, but "No file is opened recently" is even better.

Reply:
You can see what VS Code or other tools use for this. "Recent files" in general in context of editor means, recently opened files in that editor.

Reply:
> You can see what VS Code or other tools use for this.

I don't know how to completely clear the recent files in VS Code. In Sublime Text, there's no this sentence (i.e. hint for no file is opened recently). In Qt Creator, the "Open Recent Files" menu is disabled when no file is opened recently.

I still don't know what's wrong with "No file is opened recently". I think that's more clear than "No recent files".

Reply:
Nothing is wrong, it's just too long. BTW, can't we also make "Open Recent files" disabled, if no recent file is found?

Reply:
I think it's more user-friendly to display a message for the reason that Open Recent Files is not available.

Reply:
Okay, it's resolved then.